compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELinalgFunctionOutlining.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree-amd-aie/IR/AMDAIEOps.h"
+#include "iree-amd-aie/Transforms/AMDAIEUtils.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVM.h"
+#include "mlir/Conversion/FuncToLLVM/ConvertFuncToLLVMPass.h"
+#include "mlir/Dialect/Linalg/IR/LinalgInterfaces.h"
+#include "mlir/Dialect/Linalg/Transforms/Hoisting.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Dialect/Vector/IR/VectorOps.h"
+
+#define DEBUG_TYPE "iree-amdaie-linalg-function-outlining"
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+unsigned uniqueOutlinedMatmul = 0;
+unsigned uniqueOutlinedElementwise = 0;
+
+/// Utility to outline the linalg compute op.
+static FailureOr<func::FuncOp> outlinedToAFunction(
+    IRRewriter &rewriter, ModuleOp moduleOp, linalg::LinalgOp computeOp,
+    std::string outlineFuncName,
+    DenseMap<Operation *, std::string> &computeOpToOutlinedFuncMap) {
+  // Check if the compute op is equivalent to a previously outlined compute op.
+  // If yes, we replace the `outlineFuncName` of the current compute op to be
+  // same as the previous equivalent outlined compute op in order to lookup the
+  // Symbol table.
+  for (auto &[op, funcName] : computeOpToOutlinedFuncMap) {
+    if (!OperationEquivalence::isEquivalentTo(
+            computeOp.getOperation(), op,
+            OperationEquivalence::ignoreValueEquivalence, /*flags=*/nullptr,
+            OperationEquivalence::IgnoreLocations))
+      continue;
+    outlineFuncName = funcName;
+    break;
```

Comment:
Why don't you directly check if the `isEquivalentTo` condition holds, and then update outlineFuncName and break?